src/nvim/ui.c

```diff
@@ -535,14 +535,18 @@ Array ui_array(void)
 {
   Array all_uis = ARRAY_DICT_INIT;
   for (size_t i = 0; i < ui_count; i++) {
-    Dictionary dic = ARRAY_DICT_INIT;
-    PUT(dic, "width", INTEGER_OBJ(uis[i]->width));
-    PUT(dic, "height", INTEGER_OBJ(uis[i]->height));
-    PUT(dic, "rgb", BOOLEAN_OBJ(uis[i]->rgb));
+    UI *ui = uis[i];
+    Dictionary info = ARRAY_DICT_INIT;
+    PUT(info, "width", INTEGER_OBJ(ui->width));
+    PUT(info, "height", INTEGER_OBJ(ui->height));
+    PUT(info, "rgb", BOOLEAN_OBJ(ui->rgb));
```

Comment:
should `inspect()` also set these other properties?

Reply:
why? they are shared amongst all UI:s, so what would be the point of duplicating them?

Reply:
other way round then: why do we need inspect()? Is it not possible to get the channel of a UI from here?

Reply:
the channel is specific to one UI kind, the rest of the fields are generic. Other way around: why should this method deviate from how all other UI functionality delegates from general to specific implementation?

Reply:
Also I see room for futher expansion, such as terminfo info from the TUI, but that is post-0.3 to me.